PyTorch build failure with cmake 4.0.0.

cmake 4.0.0 is released on March 27, 2025. It breaks compilation of PyTorch main:

```
CMake Error at third_party/protobuf/cmake/CMakeLists.txt:2 (cmake_minimum_required):
  Compatibility with CMake < 3.5 has been removed from CMake.
  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.
  Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.
-- Configuring incomplete, errors occurred!
```

https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/14124651013/job/39571049064

--- a/.github/actions/setup-pytorch/action.yml
+++ b/.github/actions/setup-pytorch/action.yml
@@ -128,6 +128,9 @@ runs:
 
         cd pytorch
         pip install wheel
+        # FIXME: Compatibility with versions of CMake older than 3.5 has been removed, this brakes compilation of third_party/protobuf:
+        # CMake Error at third_party/protobuf/cmake/CMakeLists.txt:2 (cmake_minimum_required)
+        pip install 'cmake<4.0.0'
         pip install -r requirements.txt
         USE_XCCL=1 USE_STATIC_MKL=1 CFLAGS="-Wno-error=maybe-uninitialized" python setup.py bdist_wheel 2>&1 | grep -v \
           "Double arithmetic operation is not supported on this platform with FP64 conversion emulation mode (poison FP64 kernels is enabled)." | grep -v '^$'

